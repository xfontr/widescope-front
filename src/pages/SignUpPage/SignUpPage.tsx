import React from "react";
import SignForm from "../../components/SignForm/SignForm";

const SignUpPage = React.memo(
  (): JSX.Element => (
    <>
      <h2 className="page__title">
        <span className="page__title--bold">Sign up</span>
      </h2>

      <SignForm isLogin={false} />
    </>
  )
);

export default SignUpPage;
