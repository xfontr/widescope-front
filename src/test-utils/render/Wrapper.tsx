import { Provider } from "react-redux";
import { BrowserRouter } from "react-router-dom";
import { ThemeProvider } from "styled-components";
import { store } from "../../app/store";
import GlobalStyles from "../../styles/GlobalStyles";
import styledMainTheme from "../../styles/styledMainTheme";

interface WrapperProps {
  children: JSX.Element | JSX.Element[];
}

const Wrapper = ({ children }: WrapperProps): JSX.Element => {
  return (
    <ThemeProvider theme={styledMainTheme}>
      <Provider store={store}>
        <BrowserRouter>
          <GlobalStyles />
          {children}
        </BrowserRouter>
      </Provider>
    </ThemeProvider>
  );
};

export default Wrapper;
