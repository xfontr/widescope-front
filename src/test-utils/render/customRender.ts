import { render } from "@testing-library/react";
import { Wrapper } from "./Wrapper";

const customRender = (view: JSX.Element, ...options: any) =>
  render(view, { wrapper: Wrapper, ...options });

export * from "@testing-library/react";

export { customRender as render };
