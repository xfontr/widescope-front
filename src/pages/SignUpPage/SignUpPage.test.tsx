import { render, screen } from "../../test-utils/render/customRender";
import SignUpPage from "./SignUpPage";

describe("Given a SignUpPage component", () => {
  describe("When instantiated", () => {
    test("Then it should show a sign up form", () => {
      render(<SignUpPage />);

      const signUpHeading = screen.getByRole("heading", {
        name: "Sign up",
        level: 3,
      });

      expect(signUpHeading).toBeInTheDocument();
    });
  });
});
