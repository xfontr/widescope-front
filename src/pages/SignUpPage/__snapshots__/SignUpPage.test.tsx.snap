// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given a SignUpPage component When instantiated Then it should match the snapshot 1`] = `
Array [
  <h2
    className="page__title"
  >
    <span
      className="page__title--bold"
    >
      Sign up
    </span>
  </h2>,
  <form
    className="sc-hKMtZM ctpeAE"
    data-testid="form"
    onSubmit={[Function]}
  >
    <header
      className="sc-eCYdqJ jizDxD"
    >
      <h3
        className="form__heading"
      >
        Sign up
      </h3>
      <span
        className="form__cta-text"
      >
        Share your latests projects with the WideScope community.
      </span>
    </header>
    <div
      className="sc-jSMfEi icykpQ"
    >
      <label
        className="sc-gKXOVf jsDXqo"
        htmlFor="name"
      >
        Name
      </label>
      <input
        className="sc-iBkjds jsjeVM  form__input--user"
        id="name"
        name="name"
        onChange={[Function]}
        placeholder="John Doe"
        type="text"
        value=""
      />
    </div>
    <div
      className="sc-jSMfEi icykpQ"
    >
      <label
        className="sc-gKXOVf jsDXqo"
        htmlFor="password"
      >
        Password
      </label>
      <input
        className="sc-iBkjds jsjeVM  form__input--password"
        id="password"
        name="password"
        onChange={[Function]}
        type="password"
        value=""
      />
    </div>
    <div
      className="sc-jSMfEi icykpQ"
    >
      <label
        className="sc-gKXOVf jsDXqo"
        htmlFor="repeatPassword"
      >
        Repeat password
      </label>
      <input
        className="sc-iBkjds jsjeVM"
        id="repeatPassword"
        name="repeatPassword"
        onChange={[Function]}
        type="password"
        value=""
      />
    </div>
    <div
      className="sc-jSMfEi icykpQ"
    >
      <label
        className="sc-gKXOVf jsDXqo"
        htmlFor="email"
      >
        Email address
      </label>
      <input
        className="sc-iBkjds jsjeVM"
        id="email"
        name="email"
        onChange={[Function]}
        type="email"
        value=""
      />
    </div>
    <div
      className="sc-papXJ kLgKOk"
    >
      <ul
        className="error-list"
      />
    </div>
    <footer
      className="sc-ftvSup kVrNfS"
    >
      <button
        className="sc-bczRLJ jZmJeh undefined button--default "
        onClick={[Function]}
        rel="noopener noreferrer"
        target="_blank"
        type="submit"
      >
        Sign up
      </button>
      <span
        className="form__sign-up-cta"
      >
        Already have an account?
        <a
          href="/log-in"
          onClick={[Function]}
        >
          Log in
        </a>
      </span>
    </footer>
  </form>,
]
`;
